src/sidebar/components/test/top-bar-test.js

```diff
@@ -1,183 +1,205 @@
-        bindings: require('../login-control').bindings,
-      })
-      .component('searchInput', {
-        bindings: {
-          alwaysExpanded: '<',
-          query: '<',
-          onSearch: '&',
-        },
-      });
-  });
-
-  beforeEach(function() {
-    angular.mock.module('app', {
-      settings: fakeSettings,
-    });
+  let fakeStore;
 
+  beforeEach(() => {
     fakeIsThirdPartyService = sinon.stub().returns(false);
+    fakeStore = {
+      filterQuery: sinon.stub().returns(null),
+      setFilterQuery: sinon.stub(),
+    };
 
-    topBar.$imports.$mock({
+    TopBar.$imports.$mock({
+      '../store/use-store': callback => callback(fakeStore),
       '../util/is-third-party-service': fakeIsThirdPartyService,
     });
   });
 
   afterEach(() => {
-    topBar.$imports.$restore();
+    TopBar.$imports.$restore();
   });
 
-  function applyUpdateBtn(el) {
-    return el.querySelector('.top-bar__apply-update-btn');
+  function applyUpdateBtn(wrapper) {
+    return wrapper.find('.top-bar__apply-update-btn');
   }
 
-  function helpBtn(el) {
-    return el.querySelector('.top-bar__help-btn');
+  function helpBtn(wrapper) {
+    return wrapper.find('.top-bar__help-btn');
   }
 
-  function createTopBar(inputs) {
-    const defaultInputs = {
-      isSidebar: true,
-    };
-    return util.createDirective(
-      document,
-      'topBar',
-      Object.assign(defaultInputs, inputs)
-    );
+  function createTopBar(props = {}) {
+    const auth = { status: 'unknown' };
+    return shallow(
+      <TopBar auth={auth} isSidebar={true} settings={fakeSettings} {...props} />
+    ).dive(); // Dive through `withServices` wrapper.
   }
 
-  it('shows the pending update count', function() {
-    const el = createTopBar({
+  it('shows the pending update count', () => {
+    const wrapper = createTopBar({
       pendingUpdateCount: 1,
     });
-    const applyBtn = applyUpdateBtn(el[0]);
-    assert.ok(applyBtn);
+    const applyBtn = applyUpdateBtn(wrapper);
+    assert.isTrue(applyBtn.exists());
   });
 
-  it('does not show the pending update count when there are no updates', function() {
-    const el = createTopBar({
+  it('does not show the pending update count when there are no updates', () => {
+    const wrapper = createTopBar({
       pendingUpdateCount: 0,
     });
-    const applyBtn = applyUpdateBtn(el[0]);
-    assert.notOk(applyBtn);
+    const applyBtn = applyUpdateBtn(wrapper);
+    assert.isFalse(applyBtn.exists());
   });
 
-  it('applies updates when clicked', function() {
+  it('applies updates when clicked', () => {
     const onApplyPendingUpdates = sinon.stub();
-    const el = createTopBar({
+    const wrapper = createTopBar({
       pendingUpdateCount: 1,
-      onApplyPendingUpdates: onApplyPendingUpdates,
+      onApplyPendingUpdates,
     });
-    const applyBtn = applyUpdateBtn(el[0]);
-    applyBtn.click();
+    const applyBtn = applyUpdateBtn(wrapper);
+    applyBtn.simulate('click');
```

Comment:
I get a little confused as to when `act()` is needed and when it is not sometimes...

Reply:
Hopefully the [adapter's README](https://github.com/preactjs/enzyme-adapter-preact-pure#state-updates) clarifies the situation. The rule of thumb is that when you call a method provided by Enzyme (`mount`, `shallow`, `wrapper.{simulate, setProps}`) it handles this for you. If you call a callback prop or do something "outside" of Enzyme, you may need `act`.